operator/v1/types_ingress.go

```diff
@@ -0,0 +1,219 @@
+	// ingress controller. This is useful for implementing shards.
+	//
+	// If unset, the default is no filtering.
+	//
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// routeSelector is used to filter the set of Routes serviced by the ingress
+	// controller. This is useful for implementing shards.
+	//
+	// If unset, the default is no filtering.
+	//
+	// +optional
+	RouteSelector *metav1.LabelSelector `json:"routeSelector,omitempty"`
+
+	// nodePlacement enables explicit control over the scheduling of the ingress
+	// controller.
+	//
+	// If unset, defaults are used. See NodePlacement for more details.
+	//
+	// +optional
+	NodePlacement *NodePlacement `json:"nodePlacement,omitempty"`
+}
+
+// NodePlacement describes node scheduling configuration for an ingress
+// controller.
+type NodePlacement struct {
+	// nodeSelector is the node selector applied to ingress controller
+	// deployments.
+	//
+	// If unset, the default is:
+	//
+	//   beta.kubernetes.io/os: linux
+	//   node-role.kubernetes.io/worker: ''
+	//
+	// If set, the specified selector is used and replaces the default.
+	//
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+// EndpointPublishingStrategyType is a way to publish ingress controller endpoints.
+type EndpointPublishingStrategyType string
+
+const (
+	// LoadBalancerService publishes the ingress controler using a Kubernetes
+	// LoadBalancer Service.
+	//
+	// In this configuration, the ingress controller deployment uses container
+	// networking. A LoadBalancer Service is created to publish the deployment. If
+	// domain is set, a wildcard DNS record will point to the LoadBalancer
+	// Service's external name.
+	//
+	// See: https://kubernetes.io/docs/concepts/services-networking/#loadbalancer
+	LoadBalancerServiceStrategyType EndpointPublishingStrategyType = "LoadBalancerService"
+
+	// HostNetwork publishes the ingress controller on node ports where the
+	// ingress controller is deployed.
+	//
+	// In this configuration, the ingress controller deployment uses host
+	// networking, bound to node ports 80 and 443. The user is responsible for
+	// configuring an external load balancer to publish the ingress controller via
+	// the node ports.
+	HostNetworkStrategyType EndpointPublishingStrategyType = "HostNetwork"
+
+	// Private does not publish the ingress controller.
+	//
+	// In this configuration, the ingress controller deployment uses container
+	// networking, and is not explicitly published. The user must manually publish
+	// the ingress controller.
+	PrivateStrategyType = "Private"
+)
+
+// EndpointPublishingStrategy is the a way to publish the endpoints of an
+// IngressController, and represents the type and any additional configuration
+// for a specific type.
+type EndpointPublishingStrategy struct {
+	// type is the publishing strategy to use. Valid values are
```

Comment:
Explain them in this comment (instead of the other comment)

Reply:
Done